user/views.py

```diff
@@ -1,3 +1,51 @@
-from django.shortcuts import render
+import json, bcrypt, jwt
 
-# Create your views here.
+from django.shortcuts   import render
+from django.views       import View
+from django.http        import JsonResponse, HttpResponse
+
+from user.models        import User
+from my_settings        import SECRET_KEY,AL
+from user.utils         import login_decorator
+
+
+PASSWORD_MIN_LENGTH     = 6
+USER_NAME_MIN_LENGTH    = 3
+USER_NAME_MAX_LENGTH    = 32
+
+class SignUpInitializeView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        
+        try:
+            user_name   = data['user_name']
+            password    = data['password']
+            nickname    = data['nickname']
+            email       = data['email']
+
+            if User.objects.filter(user_name=user_name).exists():
+                return JsonResponse({'message':'USER_NAME_ALREADY_EXISTS'}, status=409)
+            if User.objects.filter(email=email).exists():
+                return JsonResponse({'message':'EMAIL_ALREADY_EXISTS'}, status=409)
+
+            if len(user_name)<USER_NAME_MIN_LENGTH:
+                return JsonResponse({'message':'SHORT_ID'}, status=400)
+            if len(user_name)>USER_NAME_MAX_LENGTH:
+                return JsonResponse({'message':'LONG_ID'}, status=409)
+            if len(password)<PASSWORD_MIN_LENGTH:
+                return JsonResponse({'message':'SHORT_PASSWORD'}, status=409)
+            
```

Comment:
```suggestion

            if User.objects.filter(user_name=user_name).exists():
                return JsonResponse({'message':'USER_NAME_ALREADY_EXISTS'}, status=409)
            if User.objects.filter(email=email).exists():
                return JsonResponse({'message':'EMAIL_ALREADY_EXISTS'}, status=409)

            if len(user_name) < USER_NAME_MIN_LENGTH:
                return JsonResponse({'message':'SHORT_ID'}, status=400)
            if len(user_name) > USER_NAME_MAX_LENGTH:
                return JsonResponse({'message':'LONG_ID'}, status=409)
            if len(password) < PASSWORD_MIN_LENGTH:
                return JsonResponse({'message':'SHORT_PASSWORD'}, status=409)
            
```
validation 꼼꼼하게 잘 하셨네요 👍 
연산자는 앞뒤로 공백을 주시는 것이 좋습니다.
그리고 user_name 길이는 두 조건을 if 문에서 동시에 판별하지 않고 따로 보내시는 것은, validation 때문인가요?

Reply:
넵! message로 길고 짧은 정도를 알려주는게 좋을것같아서 그렇게 표현했습니다.